core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -618,43 +649,35 @@ private void processWorkerRemovedBlocks(MasterWorkerInfo workerInfo,
         // Continue to remove the remaining blocks.
         continue;
       }
-      LOG.info("Block {} is removed on worker {}.", removedBlockId, workerInfo.getId());
-      workerInfo.removeBlock(masterBlockInfo.getBlockId());
-      masterBlockInfo.removeWorker(workerInfo.getId());
-      if (masterBlockInfo.getNumLocations() == 0) {
-        mLostBlocks.add(removedBlockId);
+      synchronized (masterBlockInfo) {
```

Comment:
it seems we already grabbed `workerInfo`as lock, and now we grab `blockInfo` inside `workerInfo`, do we always enforce the order like this?